test/module-test.cc

```diff
@@ -55,11 +76,551 @@ TEST(module_test, detail_namespace) {
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(arr + 0, arr + 3, ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(arr, ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(vec.begin(), vec.end(), ", ")));
+  EXPECT_EQ("1, 2, 3", to_string(fmt::join(vec, ", ")));
+}
+
+TEST(module_test, time) {
+  auto time_now = std::time(nullptr);
+  EXPECT_TRUE(fmt::localtime(time_now).tm_year > 120);
+  EXPECT_TRUE(fmt::gmtime(time_now).tm_year > 120);
+  auto chrono_now = std::chrono::system_clock::now();
+  EXPECT_TRUE(fmt::localtime(chrono_now).tm_year > 120);
+  EXPECT_TRUE(fmt::gmtime(chrono_now).tm_year > 120);
+}
+
+TEST(module_test, time_point) {
+  auto now = std::chrono::system_clock::now();
+  std::string_view past("2021-05-20 10:30:15");
+  EXPECT_TRUE(past < fmt::format("{:%Y-%m-%d %H:%M:%S}", now));
+}
+
+TEST(module_test, time_duration) {
+  EXPECT_EQ("42s", fmt::format("{}", std::chrono::seconds{42}));
+  using us = std::chrono::duration<double, std::micro>;
+  EXPECT_EQ("4.2µs", fmt::format("{:3.1}", us{4.234}));
+  const auto de = std::locale("de_DE");
+  EXPECT_EQ("4.2µs", fmt::format(de, "{:L}", us{4.2}));
+}
+
+TEST(module_test, weekday) {
+  const auto de = std::locale("de_DE");
+  EXPECT_EQ("Montag", std::format(de, "{:%A}", fmt::weekday(1)));
+}
+
+TEST(module_test, to_string_view) {
+  using fmt::to_string_view;
+  fmt::string_view nsv{to_string_view("42")};
+  EXPECT_EQ("42", nsv);
+  fmt::wstring_view wsv{to_string_view(L"42")};
+  EXPECT_EQ(L"42", wsv);
+}
+
+TEST(module_test, printf) {
+  EXPECT_WRITE(stdout, fmt::printf("%f", 42.123456), "42.123456");
+  EXPECT_WRITE(stdout, fmt::printf("%d", 42), "42");
+}
+
+TEST(module_test, fprintf) {
+  EXPECT_WRITE(stderr, fmt::fprintf(stderr, "%d", 42), "42");
+  std::ostringstream os;
+  fmt::fprintf(os, "%s", "bla");
+  EXPECT_EQ("bla", os.str());
+}
+
+TEST(module_test, sprintf) {
+  EXPECT_EQ("42", fmt::sprintf("%d", 42));
+}
+
+TEST(module_test, vprintf) {
+  EXPECT_WRITE(stdout, fmt::vprintf("%d", fmt::make_printf_args(42)), "42");
+}
+
+TEST(module_test, vfprintf) {
+  auto args =  fmt::make_printf_args(42);
+  EXPECT_WRITE(stderr, fmt::vfprintf(stderr, "%d", args), "42");
+  std::ostringstream os;
+  fmt::vfprintf(os, "%d", args);
+  EXPECT_EQ("42", os.str());
+}
+
+TEST(module_test, vsprintf) {
+  EXPECT_EQ("42", fmt::vsprintf("%d", fmt::make_printf_args(42)));
+}
+
+TEST(module_test, color) {
+  auto fg_check = fg(fmt::rgb(255, 200, 30));
+  auto bg_check = bg(fmt::color::dark_slate_gray) | fmt::emphasis::italic;
+  auto emphasis_check = fmt::emphasis::underline | fmt::emphasis::bold;
+  EXPECT_EQ("\x1B[30m42\x1B[0m",
+    fmt::format(fg(fmt::terminal_color::black), "{}", 42));
+}
+
+TEST(module_test, cstring_view) {
+  fmt::cstring_view nsv("fmt");
+  EXPECT_EQ("fmt", nsv.c_str());
+  fmt::wcstring_view wsv(L"fmt");
+  EXPECT_EQ(L"fmt", wsv.c_str());
+}
+
+TEST(module_test, buffered_file) {
+  EXPECT_TRUE(fmt::buffered_file{}.get() == nullptr);
+}
+
+TEST(module_test, move) {
```

Comment:
This test should probably be called `output_file`, not `move`.

Reply:
Copy & waste strikes again.